test/test_overrides.py

```diff
@@ -0,0 +1,756 @@
+import torch
+import numpy as np
+import unittest
+import six
+import inspect
+import pprint
+
+from common_utils import TestCase
+
+HANDLED_FUNCTIONS = {}
+HANDLED_FUNCTIONS_SUB = {}
+HANDLED_FUNCTIONS_SUB_DIAGONAL = {}
+HANDLED_FUNCTIONS_TENSOR_LIKE = {}
+
+def implements(torch_function):
+    "Register an implementation of a torch function for a Tensor-like object."
+    def decorator(func):
+        HANDLED_FUNCTIONS[torch_function.__name__] = func
+        return func
+    return decorator
+
+def implements_sub(torch_function):
+    "Register an implementation of a torch function for a Tensor-like object."
+    def decorator(func):
+        HANDLED_FUNCTIONS_SUB[torch_function.__name__] = func
+        return func
+    return decorator
+
+def implements_sub_diagonal(torch_function):
+    "Register an implementation of a torch function for a Tensor-like object."
+    def decorator(func):
+        HANDLED_FUNCTIONS_SUB_DIAGONAL[torch_function.__name__] = func
+        return func
+    return decorator
+
+def implements_tensor_like(torch_function):
+    "Register an implementation of a torch function for a Tensor-like object."
+    def decorator(func):
+        HANDLED_FUNCTIONS_TENSOR_LIKE[torch_function.__name__] = func
+        return func
+    return decorator
+
+class ImplementationMeta(type):
+    def __new__(mcs, names, bases, attrs, **kwargs):
```

Comment:
Can you comment on why you're using __new__ over __init__?

Reply:
Also, not *args, **kwargs?

Reply:
Don't metaclasses need to override `__new__`. I'm using this specific signature because it's the signature of `type.__new__`. I guess I cold use `*args, **kwargs` if that would be clearer, it doesn't really matter what the signature is.